src/cc/frontends/clang/CMakeLists.txt

```diff
@@ -1,5 +1,18 @@
 # Copyright (c) PLUMgrid, Inc.
 # Licensed under the Apache License, Version 2.0 (the "License")
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DKERNEL_MODULES_DIR='\"${BCC_KERNEL_MODULES_DIR}\"'")
+if(NOT DEFINED BCC_CUR_CPU_IDENTIFIER)
+  execute_process(COMMAND uname -r OUTPUT_VARIABLE CUR_KERNEL_VERSION)
+  # Minimum Kernel version to support BPF_F_CURRENT_CPU flag
+  execute_process(COMMAND echo "${CUR_KERNEL_VERSION}\n4.7.0"
+                  COMMAND sort -V COMMAND tail -1
+                  OUTPUT_VARIABLE HIGHER_KERNEL_VERSION)
+  if(HIGHER_KERNEL_VERSION STREQUAL CUR_KERNEL_VERSION)
+    set(BCC_CUR_CPU_IDENTIFIER "BPF_F_CURRENT_CPU")
+  else()
+    set(BCC_CUR_CPU_IDENTIFIER "bpf_get_smp_processor_id()")
+  endif()
+endif()
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DKERNEL_MODULES_DIR='\"${BCC_KERNEL_MODULES_DIR}\"' -DCUR_CPU_IDENTIFIER='\"${BCC_CUR_CPU_IDENTIFIER}\"'")
```

Comment:
We could automate this completely inside the rewriter.
Basically, rewriter can generate code like below:
```
#ifdef BPF_F_CURRENT_CPU
  bpf_perf_event_output(..., BPF_F_CURRENT_CPU, ...)
#else 
  bpf_perf_event_output(..., bpf_get_smp_processor_id(), ...)
#endif 
```

This change will be automatically picked up and compiled during the transformation from rewritten src to llvm IR.